optimum/habana/transformers/generation/utils.py

```diff
@@ -380,7 +424,7 @@ def generate(
 
         if generation_config.static_shapes:
             # token_idx is the current index in the generation process, it is incremented each time a new token is generated
-            model_kwargs["token_idx"] = torch.tensor(inputs_tensor.shape[-1], device=inputs_tensor.device)
+            model_kwargs["token_idx"] = torch.tensor(1, device=inputs_tensor.device)
```

Comment:
Why doing this? It will work only if the input is one token.

Reply:
@regisss  
Current static_shapes implementation always start with token_idx starting with 1.
inputs_tensor is already padded according to max_length.

So, everytime we increment token_idx with +1 and use index_copy and index_add for eliminating cat